### DISCUSSION TITLE
cfclient.gui connection as a USB interface

### DISCUSSION THREAD

**Discussion by shlee853** [bitcraze/discussions#1314] [Q&A]
Dear,

I'm asking about connection client.gui program using USB device interface.
Almost examples on the website are based on PA radio connection. But, I'm debugging on the USB CDC.

How to input uri for USB? when I scan device, I found usb ://0. but, when I connect usb://0, I got a error 110 message(time out).

Probably, I don't understand handshake betweek crazyfirmware and client-python program.

Could you please help me to debug handshaking using USB interface? Also, How can I debug client-python program? Is it based on QT creator IDE?

Thanks,

**Comment by gemenerik** [comment]
> I found usb ://0. but, when I connect usb://0, I got a error 110 message(time out).

You should be able to connect to the Crazyflie over USB like this. Can you share the terminal output of the error? What OS are you on? How did you install the client? Do you have the correct USB permissions?

> How can I debug client-python program?

One way of debugging the client is [from an IDE](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/installation/install/#debugging-the-client-from-an-ide)
